microservices/coco-attestation/attestation-core/src/main/java/com/suse/coco/attestation/AttestationQueueProcessor.java

```diff
@@ -0,0 +1,246 @@
+        batchSize = configurationIn.getBatchSize();
+    }
+
+    /**
+     * Create an attestation queue processor.
+     * @param sessionFactoryIn the session factory
+     * @param serviceIn the attestation result service
+     * @param executorServiceIn the executor to perform the process
+     * @param moduleLoaderIn the attestation module loader
+     * @param batchSizeIn the batch size
+     */
+    protected AttestationQueueProcessor(
+            SqlSessionFactory sessionFactoryIn,
+            AttestationResultService serviceIn,
+            ExecutorService executorServiceIn,
+            AttestationModuleLoader moduleLoaderIn,
+            int batchSizeIn
+    ) {
+        sessionFactory = sessionFactoryIn;
+        service = serviceIn;
+        executorService = executorServiceIn;
+        moduleLoader = moduleLoaderIn;
+        batchSize = batchSizeIn;
+    }
+
+    @Override
+    public void run() {
+        Thread listenerThread = createListenerThread();
+
+        try {
+            while (!Thread.currentThread().isInterrupted() && listenerRunning.get()) {
+                List<Long> results = service.getPendingResultByType(moduleLoader.getSupportedResultTypes(), batchSize);
+                if (results.isEmpty()) {
+                    LOGGER.info("No attestation result to process - Waiting");
+                    synchronized (dataAvailable) {
+                        dataAvailable.wait();
+                    }
+                    continue;
+                }
+
+                LOGGER.info("Processing attestation results {}", results);
+                CountDownLatch workersDone = new CountDownLatch(results.size());
+
+                results.forEach(resultId -> executorService.execute(() -> {
+                    try {
+                        service.processAttestationResult(resultId, (session, result) -> {
+                            AttestationWorker worker = moduleLoader.createWorker(result.getResultType());
+                            return worker.process(session, result);
+                        });
+                    }
+                    catch (Exception ex) {
+                        LOGGER.error("Unable to correctly process attestation result with id {}", resultId, ex);
+                    }
+                    finally {
+                        workersDone.countDown();
+                    }
+                }));
+
+                workersDone.await();
+            }
+        }
+        catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+        }
+        catch (Exception ex) {
+            LOGGER.error("Unable to process", ex);
+        }
+
+        // Shut down the executor
+        shutdownExecutor();
+
+        // Interrupt the listener thread if still alive
+        stopListenerThread(listenerThread);
+    }
+
+    private Thread createListenerThread() {
+        Thread listenerThread = new Thread(this::listenForAttestationResults, "attestation-processor-listener");
```

Comment:
In here we are starting a new thread to listen to changes and notify. 
What will happen if for any reason this thread dies? Will the program wait forever for a notification and stop processing stuff silently?

Reply:
If the listener thread dies, the main thread should stop as [the loop is guarded by the `listenerRunning` flag](https://github.com/uyuni-project/uyuni/blob/7c04887039ba03ca2cfa110ae416068786dd083c/microservices/coco-attestation/attestation-core/src/main/java/com/suse/coco/attestation/AttestationQueueProcessor.java#L114) which is [toggled when the listener stops](https://github.com/uyuni-project/uyuni/blob/7c04887039ba03ca2cfa110ae416068786dd083c/microservices/coco-attestation/attestation-core/src/main/java/com/suse/coco/attestation/AttestationQueueProcessor.java#L237-L238). Also there is [a test case checking that](https://github.com/uyuni-project/uyuni/blob/7c04887039ba03ca2cfa110ae416068786dd083c/microservices/coco-attestation/attestation-core/src/test/java/com/suse/coco/attestation/AttestationQueueProcessorTest.java#L99-L127). 

I have to  expand the coverage of this class though, to verify all the possible edge cases betweem the two threads.